services/config/service_test.go

```diff
@@ -1033,67 +1033,76 @@ func TestService_GetConfig(t *testing.T) {
 			},
 		},
 	}
-	for _, tc := range testCases {
+	for i, tc := range testCases {
 		updates := make(chan config.ConfigUpdate, len(testCases))
 		service, server := OpenNewSerivce(testConfig, updates)
 		defer server.Close()
 		defer service.Close()
 		basePath := server.Server.URL + httpd.BasePath + "/config"
-		// Apply all updates
-		for _, update := range tc.updates {
-			errC := make(chan error, 1)
-			go func() {
-				// Validate we got the update over the chan.
-				// This keeps the chan unblocked.
-				timer := time.NewTimer(100 * time.Millisecond)
-				defer timer.Stop()
+		if tc.expName == "" {
+			tc.expName = fmt.Sprintf("test_%d", i)
+		}
+		t.Run(tc.expName, func(t *testing.T) {
+
+			// Apply all updates
+			for _, update := range tc.updates {
+				errC := make(chan error, 1)
+				go func() {
+					// Validate we got the update over the chan.
+					// This keeps the chan unblocked.
+					timer := time.NewTimer(100 * time.Millisecond)
+					defer timer.Stop()
+					select {
+					case cu := <-updates:
+						cu.ErrC <- nil
+						errC <- nil
+					case <-timer.C:
+						errC <- errors.New("expected to get config update")
+					}
+				}()
+				resp, err := http.Post(basePath+update.Path, "application/json", strings.NewReader(update.Body))
+				if err != nil {
+					t.Fatal(err)
+				}
+				defer resp.Body.Close()
+				body, err := ioutil.ReadAll(resp.Body)
+				if err != nil {
+					t.Fatal(err)
+				}
+				if got, exp := resp.StatusCode, http.StatusNoContent; got != exp {
+					t.Fatalf("update failed: got: %d exp: %d\nBody:\n%s", got, exp, string(body))
+				}
+				timeOut := time.NewTimer(5 * time.Second)
```

Comment:
I'm always leery of tests that depend on timers, because they operate differently in different environments, but I don't have any good way to avoid it here.

Reply:
I don't either :-( This is just to reduce the fake-fail by letting it happen out of order sometimes.